Civi/Api4/Utils/FormattingUtil.php

```diff
@@ -126,7 +126,7 @@ public static function formatInputValue(&$value, ?string $fieldPath, array $fiel
     }
 
     // Special handling for 'current_user' and user lookups
-    if ($fk === 'Contact' && isset($value) && !is_numeric($value)) {
+    if ($fk === 'Contact' && isset($value) && !is_numeric($value) && !str_contains($operator, 'REGEXP')) {
```

Comment:
Test fails are because $operator is an optional param and `str_contains()` doesn't accept `null`.
But I think a whitelist would be better here because this special handling only makes sense in the context of exact-match operators. How about: